src/generator/AutoRest.Go/TemplateModels/ServiceClientTemplateModel.cs

```diff
@@ -39,10 +39,9 @@ public ServiceClientTemplateModel(ServiceClient serviceClient, string packageNam
                                              : (MethodGroupName + "Client").TrimPackageName(PackageName);
             MethodScope = new MethodScopeProvider();
             MethodTemplateModels = new List<MethodTemplateModel>();
-
             Methods.Where(m => m.BelongsToGroup(MethodGroupName, PackageName))
-                    .OrderBy(m => m.Name)
-                    .ForEach(m => MethodTemplateModels.Add(new MethodTemplateModel(m, ClientName, PackageName, new MethodScopeProvider())));
+                .OrderBy(m => m.Name)
+                .ForEach(m => MethodTemplateModels.Add(new MethodTemplateModel(m, ClientName, PackageName, new MethodScopeProvider(), m.NextMethodExists(Methods))));
                 
```

Comment:
why you need to pass m.NextMethodExists(Methods) here? Can't you check this inside new MethodTemplateModel() since it will have the information you needed to nextlink?